### DISCUSSION TITLE
nativemem profile session version from malloc branch shows more samples than expected

### DISCUSSION THREAD

**Discussion by weterock** [async-profiler/async-profiler#988] [Q&A] [answered]
Hello @apangin and thank you for such a nice feature. This helped me a lot to identify the issue https://mail.openjdk.org/pipermail/nio-dev/2015-December/003420.html really appreciate.However may I ask 4 questions and it is nice if you can help:
1. I am not sure if I completely understand what is the sample. I thought it is like 1 sample = 1 call in case of nativemem profiling however seems like it is not. 
2. Is there a way to add a time when certain sample was recorded, I understand that then it will likely can not be shown in flamegraph view, given that 1000 of samples have different time, but overall it might be nice feature like what you did with recording per thread with -t flag. 
3. I noticed that even after free was called for the same address through Unsafe, async-profiler still shows me stacktrace in flamegraph at the end. Is it somehow related to the problem that profiler makes a record in batches not per 1 sample ? 
4. In original thread you mentioned that you do not plan to merge it to master, and want to make it a bit different.
https://github.com/async-profiler/async-profiler/discussions/491 
Is there any work in progress or anything where you might need some help ? I am not familiar with c++ but maybe can share with someone who would like to contribute. 

Thanks in advance.

**Comment by apangin** [comment, marked as answer]
1 sample = 1 allocation that has not been freed by the end of the profiling session, unless a different sampling interval has been set with `interval` option.

Original timestamps are preserved in .jfr format.

Re merging the feature in master: https://github.com/async-profiler/async-profiler/discussions/836#discussioncomment-10055271

**Reply by weterock** [reply]
Thanks a lot. Will retry with jfr format once again.

**Reply by weterock** [reply]
Tried to check with jfr format and everything became super clear now
Great instrument.
